sentry-core/src/client.rs

```diff
@@ -43,7 +43,7 @@ pub(crate) type TransportArc = Arc<RwLock<Option<Arc<dyn Transport>>>>;
 pub struct Client {
     options: ClientOptions,
     transport: TransportArc,
-    session_flusher: SessionFlusher,
+    session_flusher: RwLock<Option<SessionFlusher>>,
```

Comment:
Would it be possible to somehow leave an implementation note that explains why this is an option?  If I read this correctly it only is an option because it allows you to explicitly drop the SessionFlusher on close.  So this rather different from the thing being actually optional as you'd expect from an option.

Reply:
agreed. Although maybe it would be better to just make `client` a newtype around `Option<ClientInner>` or something.

Reply:
that sounds even nicer, also more work though ;)